### DISCUSSION TITLE
The story of Symfony\Component\Console\Terminal

### DISCUSSION THREAD

**Discussion by feamsr00** [symfony/symfony#49004] [Q&A]
The short of it. Terminal has some great methods. Between Terminal and Cursor, I feel that one could make some great console apps with Symfony Console. (Drawing in arbitrary locations with Cursor is great and all, but Terminal allows you to know the "size and shape" of your *canvas*) That said, I saw that [Terminal made an appearance on Application](https://symfony.com/blog/new-in-symfony-3-2-console-improvements-part-2#introduced-a-new-terminal-class) through a getter and then [just as quickly it disappeared](https://github.com/symfony/symfony/blob/030514f5770401e822f91069ffb9fc5bb3289506/src/Symfony/Component/Console/CHANGELOG.md#400). Cursor has [a whole page dedicated](https://symfony.com/doc/current/components/console/helpers/cursor.html) to this rather neat helper. However not a mention of Terminal anywhere. 

Am I reading too much into this? Is Terminal fully supported, but just under documented? Or is it "[Seward’s Folly](https://www.history.com/news/why-the-purchase-of-alaska-was-far-from-folly)"

**Comment by MatTheCat** [comment]
`Application::getTerminal` never got public: it was added and removed in the same PR (#19012).

Methods like `Application::getTerminalDimensions` were removed since you could get these yourself through `Terminal`, and I guess `Application::setTerminalDimensions` was removed because “playing with the dimensions should only be done when testing an app” (https://github.com/symfony/symfony/pull/19012#issuecomment-224930425).

It’s true `Terminal` never got documented; maybe because there aren’t many use-cases in userland? You can still open a PR on [symfony-docs](https://github.com/symfony/symfony-docs)!

**Reply by feamsr00** [reply]
Great points. Under documented it seems. I'm sure we'll see some interesting use cases sometime soon!
